straxbra/plugins.py

```diff
@@ -372,28 +376,50 @@ def compute(self, peaks):
                  default=100)
 )
 class PeakPositions(strax.Plugin):
-    """
-    I actually wrote this
+    '''
+    Position Reconstruction for XeBRA
+    
     Version 0.0.1: weighted sum
-    Version 0.0.2: hopefully LRF or better
-    """
-    __version__ = "0.0.1"
+    Version 0.0.2: LRF
+    
+    Status: July 2019, Version 0.0.2
+    
+    Position reconstruction for XeBRA following the position reconstruction algorithm of Mercury 
+    (employed in the LUX experiment, originally developed for the ZEPLIN-III dark matter experiment) 
+    with light response functions (see https://arxiv.org/abs/1710.02752v2 and 
+    https://arxiv.org/abs/1112.1481 ).
+    '''
+    __version__ = "0.0.2"
     dtype = [('x', np.float32,
-              'Reconstructed S2 X position (cm), uncorrected'),
+              'Reconstructed S2 X position (mm), uncorrected'),
              ('y', np.float32,
-              'Reconstructed S2 Y position (cm), uncorrected')]
+              'Reconstructed S2 Y position (mm), uncorrected')]
     depends_on = ('peaks',)
 
     parallel = False
 
     def setup(self):
-
+        # PMT mask
         self.pmt_mask = np.zeros_like(self.config['to_pe'], dtype=np.bool)
         self.pmt_mask[self.config['top_pmts']] = self.config['to_pe'][self.config['top_pmts']] > 0
-        pmt_x = np.array([-14.,-28,-14.,14.,28.,14.,0.])
-        pmt_y = np.array([-28.,0.,28.,28.,0.,-28.,0.])
+        # PMT positions in mm in cartesian coordinates
+        pmt_x = np.array([-14.,-28.,-14.,14.,28.,14.,0.]) # x-positions PMTs
+        pmt_x = pmt_x[self.pmt_mask]
+        pmt_y = np.array([-28.,0.,28.,28.,0.,-28.,0.])    # y-positions PMTs
+        pmt_y = pmt_y[self.pmt_mask]
         self.pmt_positions = np.column_stack((pmt_x, pmt_y))
-
+        # Fit parameters LRFs
+        # MC driven (R_PTFE = 95 %, T_meshes = 89.770509 %, lambda_LXe = 100 cm); 
+        # To be iteratively determined from data later
+        self.fitparameters = np.array([np.array([ 0.58534179, 29.89341846, -0.3275816, 4.14715081, 2.61234684]), 
+                                  np.array([ 0.63546614, 28.49525342, -0.19651583, 3.76011493, 2.68672152]), 
+                                  np.array([ 0.58840586, 30.39015033, -0.38248759, 4.27642323, 2.56774842]), 
+                                  np.array([ 0.59111988, 31.38350968, -0.5760228, 4.64122418, 2.51311592]), 
+                                  np.array([ 0.63771524, 28.90663204, -0.26194541, 3.90052756, 2.66510948]), 
+                                  np.array([ 0.59030322, 30.55082687, -0.46183924, 4.39769959, 2.55446814]), 
+                                  np.array([ 0.53114467, 39.20861977, -17.93187819, 20.60397171, 2.27692367])])
```

Comment:
Is it strictly necessary to have `np.array([np.array([...]), np.array([...]),...])`? Can't we do `np.array([[...],[...]...])`?

Reply:
Nope, changed it to the simplified notation, the `np.array`s inside are just a relic from the manual copy of the fit parameters from the simulations.